### PR TITLE
GetStats: return source url

### PR DESCRIPTION
**Pull request by shifucun**
This will be used by client side to show data source url.

This PR also use getToken() function in memcache parallel read to align with BT read.

### CODE REVIEW

File: proto/mixer.proto [hunk 1]
```diff
@@ -190,13 +190,15 @@ message ObsTimeSeries {
     string measurement_method = 3;
     string observation_period = 4;
     string import_name = 5;
+    string provenanceDomain = 6;
```
Comment (pradh) [inline proto/mixer.proto:193]: This deviates from the use of snake case above.

File: proto/mixer.proto [hunk 2]
```diff
@@ -190,13 +190,15 @@ message ObsTimeSeries {
...
   }
   map<string, double> data = 1;  // Date to value.
   string unit = 2;
   string place_name = 3;
   bool is_dc_aggregate = 4;
   string place_dcid = 5;  // Only set if the key uses mid.
   repeated SourceSeries source_series = 6;
+  string provenanceDomain = 7;
```
Comment (pradh) [inline proto/mixer.proto:201]: Puzzled why this is needed since you have it in SourceSeries?

Comment (shifucun) [reply proto/mixer.proto:201]: Now i only return one source series, and this is popped to upper level.

File: proto/mixer.proto (hunk 1 above)
Comment (shifucun) [reply proto/mixer.proto]: changed.

File: proto/mixer.proto (hunk 2 above)
Comment (pradh) [reply proto/mixer.proto]: Is it temporary then? Add this comment to proto and have a TODO to remove if temp (later PR is fine).

**Pull request merged**